flink-connectors/flink-connector-jdbc/src/main/java/org/apache/flink/connector/jdbc/internal/executor/InsertOrUpdateJdbcExecutor.java

```diff
@@ -83,6 +88,21 @@ public void open(Connection connection) throws SQLException {
 		updateStatement = connection.prepareStatement(updateSQL);
 	}
 
+	@Override
+	public void reopen(Connection connection) throws SQLException {
+		try {
+			existStatement.close();
+			insertStatement.close();
+			updateStatement.close();
+		} catch (SQLException e) {
+			LOG.info("PreparedStatement close failed.", e);
+		}
+
+		existStatement = connection.prepareStatement(existSQL);
+		insertStatement = connection.prepareStatement(insertSQL);
+		updateStatement = connection.prepareStatement(updateSQL);
```

Comment:
I think we don't need to introduce a new `reopen` interface, the implementation is always the combination of `close` and `open`. You can just call `open` and then `close`. 

But maybe we need to move the construction of `batch` map into constuctor. This is also better because we can mark `batch` as `final` then. We can also rename `open` to `openConnection` and `close` to `closeConnection` to make the behavior more explicitly.

Reply:
Thanks for review. Yes, move the `batch` map into constructor looks better.  However, I think it will confuse people if change `open` to `openConnection`. Because in `open`, we just init `PrepareStatement`, and at this moment the JDBC connection is already open. `openConnection` will confuse people with "open JDBC connecton".

Reply:
Hi @wuchong , I have checked the implementation of `open` and `close`. The `reopen` is not the simple combination of `close` and `open`. The difference is that `close` will clear the `batch` map. It's not what we wan't. 

So we need to introduce a new `reopen` interface.

Reply:
> Thanks for review, @wuchong . Yes, move the `batch` map into constructor looks better. However, I think it will confuse people if change `open` to `openConnection`. Because in `open`, we just init `PrepareStatement`, and at this moment the JDBC connection is already open. `openConnection` will confuse people with "open JDBC connecton".

Then I suggest to call them `prepareStatements(Connection)` and `closeStatements(Connection)`.


> Hi @wuchong , I have checked the implementation of `open` and `close`. The `reopen` is not the simple combination of `close` and `open`. The difference is that `close` will clear the `batch` map. It's not what we wan't.
> 
> So we need to introduce a new `reopen` interface.

I know currently we have an additional `batch` map initializatino in the `open`, that's why I suggest to move it into constructor in the previous comment.

Reply:
> Then I suggest to call them `prepareStatements(Connection)` and `closeStatements(Connection)`.

OK, I will rename them.